File: leetcode/3315. Construct the Minimum Bitwise Array II/main.py
from typing import List


class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        result = []

        for num in nums:
            if num != 2:
                result.append(num - ((num + 1) & (-num - 1)) // 2)
            else:
                result.append(-1)

        return result

File: leetcode/3315. Construct the Minimum Bitwise Array II/main.ts
const minBitwiseArray = (nums: number[]): number[] => {
  return nums.map((num, idx) => {
    if (num != 2) {
      return (num - ((num + 1) & (-num - 1)) / 2);
    }

    return -1;
  })
};
